azurelinuxagent/common/protocol/extensions_goal_state.py

```diff
@@ -100,9 +100,9 @@ def compare(from_extensions_config, from_vm_settings):
         context = []  # used to keep track of the attribute that is being compared
 
         def compare_goal_states(first, second):
+            compare_attributes(first, second, "created_on_timestamp")
             compare_attributes(first, second, "activity_id")
             compare_attributes(first, second, "correlation_id")
-            compare_attributes(first, second, "created_on_timestamp")
```

Comment:
Would it be worth leaving a comment here mentioning the importance of the ordering here?

Reply:
it's a good point, but all this code will be ripped off in the next iteration (which is already happening on a separate branch) and currently I do not have an environment where i can run unit tests to verify changes do this PR i will limit changes to functional/correctness issues

Reply:
I had to fix some typos, so I went ahead and added the comment. Thanks for pointing it out.